frontend/src/pages/project/Snapshots/index.vue

```diff
@@ -73,12 +63,30 @@ export default {
     },
     methods: {
         fetchData: async function (newVal) {
-            this.loading = true
             if (this.project.id) {
+                this.loading = true
+                const deviceCounts = await this.countDevices()
                 const data = await snapshotApi.getProjectSnapshots(this.project.id)
-                this.snapshots = data.snapshots
+                this.snapshots = data.snapshots.map((s) => {
+                    s.deviceCount = deviceCounts[s.id]
+                    return s
+                })
+                this.loading = false
             }
-            this.loading = false
+        },
+        async countDevices () {
+            // hardcoded device limit to ensure all are returned - feels dirty
+            const data = await devicesApi.getDevices(null, 10000000)
```

Comment:
I'm going to move this server-side as I agree it feels dirty and excessive just to count snapshots!

Reply:
The devices API only works for **platform** admins. As such this needs to be **moved** server side.